### PR TITLE
Implement TimeTableByWeek

### PR DESCRIPTION
**Pull request by eleidan**
The main idea is to use day-like view without weekends.
Vacation type is marked with its particular color.
Vacation with the 'requested' status is marked with linear gradient.

Integrate **Moment.js** and **moment-range**

Add `getWeekRange()` helper

Update HTML layout for master page

Fix collections fetch logic:

  The `drawEmptyTable()` method inside `renderVacationsTable()` method
  relays on a list of team members.

  So there must be the following synchronous fetching order:
    1. Fetch list of members.
    2. Fetch list of vacations.

  This fix ensures that the list of members is fetched and ready
  before fetching vacations list.

### CODE REVIEW

File: app/assets/javascripts/collections/team_members.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 App.Collections.TeamMembers = Backbone.Collection.extend({
+  model: App.Models.TeamMember,
+
   initialize: function( teamID ) {
     this.url = function() {
       return '/team_members/' + teamID;
     };
-  }
+  },
```
Comment (afurm) [inline app/assets/javascripts/collections/team_members.js]: Please, delete comma.

File: app/assets/javascripts/views/time_tables.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ App.Views.TimeTable = Backbone.View.extend({
   render: function() {
     this.renderTemplate();
     this.renderTeamMembersList();
-    this.renderTimeTableByDay();
+    // this.renderTimeTableByDay();
```
Comment (afurm) [inline app/assets/javascripts/views/time_tables.js]: Why do we need this?

**Comment by epmlys** [comment]
+ 1

File: app/assets/javascripts/views/time_table_by_week.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        continue;
+      }
+      selector = '#'+userID+'-'+date.format('YY-MM-DD');
+      $(selector).addClass(vacation.kind);
+      $(selector).addClass(vacation.status);
+    }
+  },
+
+
+  // ***************************** Helpers *************************************
```
Comment (afurm) [inline app/assets/javascripts/views/time_table_by_week.js]: Maybe it's a good idea to move helpers into separate file. What do you think?

File: app/assets/javascripts/views/time_tables.js [hunk 4]
```diff
@@ -61,7 +62,6 @@ App.Views.TimeTable = Backbone.View.extend({
     }, this);
   },
 
-  // *************************************************************************
   updateTimeTableDateRange: function() {
     // code here
```
Comment (afurm) [inline app/assets/javascripts/views/time_tables.js]: What does this mean?

File: app/assets/javascripts/collections/team_members.js (hunk 1 above)
Comment (eleidan) [inline app/assets/javascripts/collections/team_members.js]: Done.

File: app/assets/javascripts/views/time_tables.js (hunk 2 above)
Comment (eleidan) [inline app/assets/javascripts/views/time_tables.js]: I removed commented code.

File: app/assets/javascripts/views/time_table_by_week.js (hunk 3 above)
Comment (eleidan) [inline app/assets/javascripts/views/time_table_by_week.js]: I prefer having them right here and have benefits from using view's this context.

Comment (andriiNyzhnik) [inline app/assets/javascripts/views/time_table_by_week.js]: You can set view's `this` as an argument to helper method.

Comment (eleidan) [inline app/assets/javascripts/views/time_table_by_week.js]: Let me explain a bit :)

The `updateTimeTableDateRange()` method is a helper for the draft solution only. It is to be removed. The date range will be passed in by `TimeTables` view as options.

The `getNumberOfWeeks()` method is for the local use only as well as `calculateTableWidth()`.

There must be the `TimeTableByDay` view. I suppose it should have similar methods.

I plan refactoring after `Dashboard` is ready, and extract all the helpers in scope of this major refactoring.

**Pull request merged**